src/main/java/internet/shop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,184 @@
+package internet.shop.dao.jdbc;
+
+import internet.shop.dao.UserDao;
+import internet.shop.exceptions.DataProcessingException;
+import internet.shop.lib.Dao;
+import internet.shop.model.Role;
+import internet.shop.model.User;
+import internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Dao
+public class UserDaoJdbcImpl implements UserDao {
+    @Override
+    public Optional<User> findByLogin(String login) {
+        String query = "SELECT * FROM users WHERE login = ? AND deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
```

Comment:
if you are closing the statement manually, there is no need put it in try-with-resources

Reply:
Agree, fixed :)